scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin.lua

```diff
@@ -16,9 +17,21 @@ function onMobInitialize( mob )
 end
 
 -----------------------------------
+-- onMobSpawn Action
+-----------------------------------
+function onMobSpawn(mob)
+    mob:setMod(MOD_WINDRES, -64);
+end
+
+-----------------------------------
 -- onMobFight Action
 -----------------------------------
 function onMobFight( mob, target )
+
+    if (mob:getHPP() < math.random(50,60) and mob:getLocalVar("astralFlow") == 0) then
+        -- Spawn Avatar
+        mob:spawnPet();

```

Comment:
Kirin spawns the pet.  The pet uses the appropriate ability based on the selected avatar model.